### PR TITLE
Bulk Import resource tweaks

### PR DESCRIPTION
**Pull request by actlikewill**
- Adds Skip Widget for skipping rows
- Adds source file widget for validating source files
- Refactors getting case numbers and source file logic into class methods
- Fixes tests

### CODE REVIEW

**Review by longhotsummer — COMMENTED** [review]
Can we just use `requests.get`? It handles all the correct URL escaping. This feels way too much like we're having to care about things we shouldn't have to worry about.

File: peachjam/download.py [hunk 1]
```diff
@@ -57,9 +57,10 @@ def download_file_from_google(file_id):
...
     try:
         f = NamedTemporaryFile()
-        urllib.request.urlretrieve(url, f.name)
-    except urllib.error.HTTPError as e:
+        r = requests.get(url)
+        r.raise_for_status()
+        f.write(r.content)
+        return f
+    except requests.exceptions.HTTPError as e:
         logger.error(e)
```
Comment (longhotsummer) [inline peachjam/download.py:65]: Surely downloading the file is a key part of the import? If that fails, the import shouldn't succeed, I would think?

Comment (actlikewill) [reply peachjam/download.py:65]: @longhotsummer I have been grappling with what to do here. The issue is that the whole import fails if only one file is missing and the error is not shown to the user. Should we fail the whole import if one file is missing?

File: peachjam/resources.py [hunk 2]
```diff
@@ -44,6 +51,25 @@ def clean(self, value, row=None, *args, **kwargs):
         return None
 
 
+class SourceFileWidget(CharWidget):
+    def clean(self, value, row=None, **kwargs):
+        try:
+            print(value)
```
Comment (longhotsummer) [inline peachjam/resources.py:57]:
```suggestion

```

**Comment by actlikewill** [comment]
@longhotsummer Changes made.

**Review by longhotsummer — APPROVED** [review]

**Pull request merged**